server/uploader.js

```diff
@@ -81,84 +61,130 @@ module.exports = {
       }
-      seg[chName].push({
-        x: xl,
-        val: val
-      });
-      lastxr = xr;
-    }
-
-    // write to *.bcwig file
-    var namecode = bwFile.substr(0, bwFile.length - 3);
-
-    // if the folder already exists, then delete it
-    if (fs.exists(prefix + namecode)) {
-      fs.rmdirSync(prefix + namecode);
-      console.log('Wiggle file ' + namecode + ' is replaced.');
-    }
-    fs.mkdir(prefix + namecode);
-
-    for (var chr in seg) {
-      var bcwigFile = prefix + namecode + '/' + namecode + '_' + chr + '.bcwig';
-      for (var i = 0; i < seg[chr].length; i++) {
-        var bcwigBuf = new Buffer(8 * seg[chr].length);
-
-        bcwigBuf.writeInt32LE(seg[chr][i].x, i * 4);
-        bcwigBuf.writeFloatLE(seg[chr][i].val, i * 4 + 4);
+      mkdirp(folder);
+
+      for (var chr in seg) {
+        var bcwigFile = folder + '/' + bwFile + '_' + chr + '.bcwig';
         var fd = fs.openSync(bcwigFile, 'w');
-        fs.writeSync(fd, buf, 0, 4 * seg[chr].length, 0);
+        for (var i = 0; i < seg[chr].length; i++) {
+          var bcwigBuf = new Buffer(8 * seg[chr].length);
+          bcwigBuf.writeInt32LE(seg[chr][i].x, i * 4);
+          bcwigBuf.writeFloatLE(seg[chr][i].val, i * 4 + 4);
+        }
+        fs.writeSync(fd, bcwigBuf, 0, 8 * seg[chr].length, 0);
+        fs.closeSync(fd);
+      }
+
+
+      // build segment tree and save
+      for (var chr in seg) {
+        var segFile = folder + '/' + bwFile + '_' + chr + '.seg';
+        var nodes = [];
+        segtree.buildSegmentTree(nodes, seg[chr]);
+        var segBuf = new Buffer(4 + 4 * nodes.length);
+          segBuf.writeInt32LE(nodes.length, 0);
+        for (var i = 0, offset = 4; i < nodes.length; i++, offset += 2) {
+          segBuf.writeFloatLE(nodes[i], offset);
+        }
+        var fd = fs.openSync(segFile, 'w');
+        fs.writeSync(fd, segBuf, 0, offset, 0);
         fs.closeSync(fd);
       }
-    }
-
-
-    // build segment tree and save
-    for (var chr in seg) {
-      var segFile = prefix + namecode + '/' + namecode + '_' + chr + '.seg';
-      var nodes = [];
-      segtree.buildSegmentTree(nodes, seg[chr]);
-      var segBuf = new Buffer(4 + 4 * nodes.length);
-      segBuf.writeInt32LE(nodes.length, 0);
-      for (var i = 0, offset = 4; i < nodes.length; i++, offset += 2) {
-        segBuf.writeFloatLE(nodes[i], offset);
+
+    });
+
+  },
+
+  /**
+   * Sort bed data segments and write it into separate files.
+   * @param {string} prefix Folder to the bed files.
+   * @param {string} bedFile File name of the bed file.
+   */
+  bedSort: function(prefix, bedFile) {
+    var lines = rl.createInterface({
+      input: fs.createReadStream(prefix + bedFile),
+      terminal: false
+    });
+    console.log('separating bed data...');
+    var data = {};
+    lines.on('line', function(line) {
+      var parts = line.split('\t');
+      if (parts[0].indexOf('track') != -1) return;
```

Comment:
When it code gets complicated it's hard to trace what the "parts" are. It is recommended to create some variables with meaningful names and store the parts in the variables, e.g.
```js
var chrStart = parseInt(parts[1]);
var chrEnd = parseInt(parts[2]);
```
Or at least add a comment about the line format, so that someone else can see what each part is.